starter-core/src/main/java/io/micronaut/starter/feature/database/DataAzureCosmosFeature.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.database;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.build.dependencies.Priority;
+import io.micronaut.starter.feature.FeatureContext;
+import io.micronaut.starter.options.BuildTool;
+import jakarta.inject.Singleton;
+
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Add support for Micronaut Data Azure Cosmos.
+ */
+@Singleton
+public class DataAzureCosmosFeature implements DataFeature {
+
+    private static final String MICRONAUT_DATA_GROUP = "io.micronaut.data";
+    private static final String MICRONAUT_DATA_VERSION = "micronaut.data.version";
+    private static final String MICRONAUT_DATA_PROCESSOR_ARTIFACT = "micronaut-data-processor";
+    private static final String MICRONAUT_DATA_DOCUMENT_PROCESSOR_ARTIFACT = "micronaut-data-document-processor";
+    private static final String MICRONAUT_DATA_AZURE_COSMOS_DATA_ARTIFACT = "micronaut-data-azure-cosmos";
+
+    // These are just placeholder values for user to replace
+    private static final String DEFAULT_ENDPOINT = "https://some-host/some-db";
```

Comment:
I am not sure about generating configuration with placeholders. We typically don't do that for other applications and when we do we use placeholders which are valid. if we keep this, when a user generates an app. `./gradlew test` will fail with: 

```
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [io.micronaut.data.cosmos.common.CosmosDatabaseInitializer] could not be loaded: Error instantiating bean of type  [io.micronaut.data.cosmos.common.CosmosDatabaseInitializer]

Message: Client initialization failed. Check if the endpoint is reachable and if your auth token is valid. More info: https://aka.ms/cosmosdb-tsg-service-unavailable-java
Path Taken: new CosmosDatabaseInitializer() --> CosmosDatabaseInitializer.initialize([CosmosClient cosmosClient],RuntimeEntityRegistry runtimeEntityRegistry,CosmosDiagnosticsProcessor cosmosDiagnosticsProcessor,CosmosDatabaseConfiguration configuration)
	at app//io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1921)
	at app//io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:249)
	at app//io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3326)
	at app//io.micronaut.context.DefaultBeanContext.finalizeConfiguration(DefaultBeanContext.java:3684)
	at app//io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:341)
```. 

We will not be able to write a function test for this feature if the generated configuration is invalid.

Reply:
Don't know if it makes sense then without test resources support, there will never be option to have valid placeholder here in config. We have this class that triggers creation of cosmos client etc.
```
@Requires(classes = CosmosClient.class)
final class CosmosDatabaseInitializer {

 @PostConstruct
    void initialize(CosmosClient cosmosClient,
                    RuntimeEntityRegistry runtimeEntityRegistry,
                    @Nullable
                    CosmosDiagnosticsProcessor cosmosDiagnosticsProcessor,
                    CosmosDatabaseConfiguration configuration) {
```
Maybe adding 
```
@Requires(property = "azure.cosmos.endpoint")
```
would make sense and then we can create this feature without endpoint and key.

Reply:
Agree.  if `CosmosDatabaseInitializer` requires  a property to be present it should be added it as an `@Requires`. Maybe something like: 

```java
@Requires(bean= CosmosDatabaseConfiguration.class, beanProperty="endpoint")
```

Reply:
There isn't such field on `CosmosDatabaseConfiguration` so this is done via 
```
@ConfigurationBuilder(prefixes = "")
 protected CosmosClientBuilder cosmosClientBuilder = new CosmosClientBuilder();
```
so `CosmosClientBuilder` (from azure cosmos lib) gets endpoint and key properties. Don't know how to configure `@Requires` in that case but tried this `@Requires(property = "azure.cosmos.endpoint")` and it worked

Reply:
I have updated default configuration to not contain endpoint and key since makes no sense. Changes are done also in micronaut-data:azure-data-cosmos to not initialize when endpoint and key are not set. That is in version 3.9.5 now.
But don't know what else is needed to micronaut-starter to pickup correct version as now it generates this in maven for example 
```
<micronaut.data.version>3.9.4</micronaut.data.version>
```